src/OpenTelemetry.Extensions/Trace/AutoFlushActivityProcessor.cs

```diff
@@ -87,7 +87,7 @@ public override void OnEnd(Activity data)
         var shouldFlush = this.RunPredicate(data);
         if (shouldFlush)
         {
-            this.tracerProvider.ForceFlush(this.timeoutMilliseconds);
+            this.tracerProvider?.ForceFlush(this.timeoutMilliseconds);
```

Comment:
The checks above should ensure that `tracerProvider` should not be null at this line.

Reply:
Code analyzer is not smart enough to detect this so I have decided to check it once more time in case of future code changes. Changed `?` to `!`.

Fixed in a445ce05cbd21efad58aed8498a9173d355acc92